packages/mobx/src/api/annotation.ts

```diff
@@ -25,3 +33,48 @@ export type AnnotationsMap<T, AdditionalFields extends PropertyKey> = {
     [P in Exclude<keyof T, "toString">]?: AnnotationMapEntry
 } &
     Record<AdditionalFields, AnnotationMapEntry>
+
+/**
+ * Infers the best fitting annotation from property descriptor or false if the field shoudn't be annotated
+ * - getter(+setter) -> computed
+ * - setter w/o getter -> false (ignore)
+ * - flow -> false (ignore)
+ * - generator -> flow
+ * - action -> false (ignore)
+ * - function -> action (optionally bound)
+ * - other -> defaultAnnotation
+ */
+export function inferAnnotationFromDescriptor(
+    desc: PropertyDescriptor,
+    defaultAnnotation: Annotation,
+    autoBind: boolean
+): Annotation | false {
+    if (desc.get) return computed
+    if (desc.set) return false // ignore lone setter
+    // If already wrapped in action/flow, don't do that another time, but assume it is already set up properly.
+    return isFunction(desc.value)
+        ? isGenerator(desc.value)
+            ? isFlow(desc.value)
+                ? false
+                : flow
+            : isAction(desc.value)
+            ? false
+            : autoBind
+            ? autoAction.bound
+            : autoAction
+        : defaultAnnotation
+}
+
+export function isAnnotation(thing: any) {
+    return (
+        // Can be function
+        thing instanceof Object &&
+        typeof thing.annotationType_ === "string" &&
```

Comment:
The check is used only on devel for validation. Because the annotation object may need to be copied quite often I tried to keep it lean. Let me know if I should go ahead and add the symbol.

Reply:
should be ok then :)